### ISSUE TITLE
wezterm crash when cannot found default_prog without any tips on macos

### ISSUE THREAD

**Issue by cathaysia** [wez/wezterm#3928]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

WezTerm-macos-20230408-112425-69ae8472

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I set 

```lua
default_prog = { "zsh" },
```

then wezterm crash immediately and without any tips.

### Configuration

```lua
default_prog = { "zsh" },
```

### Expected Behavior

give a tips about the reason or fallback to the default shell.

**Comment by wez** [comment]
in `main`:

`wezterm -n --config 'default_prog={"woot"}'` now shows as:

![image](https://github.com/wez/wezterm/assets/117777/b57bcf70-6ef1-4c2d-bc2a-aa07581097d2)

while:

`wezterm -n --config 'default_prog={"/bin/woot"}'` now shows as:

![image](https://github.com/wez/wezterm/assets/117777/66774964-c8c1-4fba-8589-bfa141c5e841)

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by cathaysia** [comment]
I foun this [action](https://github.com/wez/wezterm/actions/runs/5501892877) already complete. and I download it from https://wezfurlong.
org/wezterm/install/macos.html#installing-on-macos . But the problem seem not fixed. it's behavior just like before:

![截屏2023-07-10 09 30 23](https://github.com/wez/wezterm/assets/42114817/9084c139-8457-4d71-bb39-6fe07b8b026b)

**Comment by cathaysia** [comment]
if I set default_prog={"/bin/woot"} or default_prog={"woot"} then I can see:

![截屏2023-07-10 09 32 14](https://github.com/wez/wezterm/assets/42114817/e6d430a5-fc03-49fc-b6ea-96a1a44b63e8)

but default_prog={"zsh"} just like before.

**Comment by wez** [comment]
> I foun this [action](https://github.com/wez/wezterm/actions/runs/5501892877) already complete. and I download it from https://wezfurlong. org/wezterm/install/macos.html#installing-on-macos . But the problem seem not fixed. it's behavior just like before:
> 
> ![截屏2023-07-10 09 30 23](https://user-images.githubusercontent.com/42114817/252187267-9084c139-8457-4d71-bb39-6fe07b8b026b.png)

Is that what you see when it is set to `zsh`?  What does the text say?  I don't that language?

**Comment by cathaysia** [comment]
that is :

```monospace
+------------------------------------+
|    "wezterm-gui" quit unexpectedly |
|  press "restart" so reopen.        |
|  press "report" to see detalis and |
|  sent report to Apple              |              
+------------------------------------+
|     reopen                         |
+------------------------------------+
|    report                          |
+------------------------------------+
|    ignore                          |
+------------------------------------+
```

when I press report, I can see some memory info:  https://gist.github.com/cathaysia/5c7152efc40f7d3cda508c6d40b11001

**Comment by wez** [comment]
If you set `default_prog={"zsh", "-n"}` does it behave differently?
What if you use the full path to `zsh`? I'm not on my mac at the moment, but I think it is: `default_prog={"/bin/zsh"}`.
Do you have alternative zsh binary installed? Maybe there is an issue with it?

**Comment by cathaysia** [comment]
no, just like before.

**Comment by cathaysia** [comment]
> If you set default_prog={"zsh", "-n"} does it behave differently?
no, just like default_prog={"zsh"}

> What if you use the full path to zsh? I'm not on my mac at the moment, but I think it is: default_prog={"/bin/zsh"}.
Yes, this is I'm use currently.

> Do you have alternative zsh binary installed? Maybe there is an issue with it?
No, this zsh is come from macos.

**Comment by cathaysia** [comment]
Interestingly, wezterm works when I set ` default_prog = { "bash" }`

**Comment by wez** [comment]
So `/bin/zsh` works but `zsh` does not?

**Comment by cathaysia** [comment]
yes, and I can create tab with zsh, just cannot boot a new wezterm.
________________________________

**Comment by wez** [comment]
re: lldb, I see the same thing here, but I don't have a crash or any other problems launching wezterm.
I think that trace is unrelated and that is some issue with lldb on arm macos.

Since you have build a debug build, can you run the debug build of wezterm outside of lldb and see if you get a better looking stack trace from the macos report dialog?

**Comment by cathaysia** [comment]
No, the report just like before: https://gist.github.com/cathaysia/7fc53a0cd4d9d0e97f4a2ea4563b8ab0

If lldb cannot be used to debug wezterm. I'm afraid there is no other means of debugging available. because gdb is not available on aarch64-darwin

**Comment by wez** [comment]
Likely blocked from updating to a version with a fix by:
* https://github.com/sfackler/rust-openssl/pull/1925

**Comment by wez** [comment]
openssl is required by the ssh libraries used in wezterm (libssh, and libssh2), so it is unlikely to go away in the foreseeable future.

**Comment by cathaysia** [comment]
This issue has been fixed in the latest commit(914f18b19cc5bab3dd04f699d9cea17bbcf74d5b), it's not clear why. I'll take the time to investigate.

**Comment by cathaysia** [comment]
This problem cannot be reproduced now, and I don't know why. But I guess it's a problem with macos itself.
